### ISSUE TITLE
Consider modifying the CoreOS host's DNS resolution

### ISSUE THREAD

**Issue by barakmich** [coreos/coreos-kubernetes#438]
Once Kubernetes is up and running, `kube-dns` is running and providing DNS to services inside of k8s on the flannel network, under `svc.cluster.local` (eg, `kube-dns.kube-system.svc.cluster.local`)*

It could be really useful for plugins, which don't have the same network namespaces as pods, to be able to resolve these services as though they were in the pod. This amounts to adding the appropriate line to `/etc/resolv.conf` during `{controller,worker}-install.sh` . This is controlled by systemd in CoreOS, so maybe `/etc/systemd/resolved.conf` is the right thing to edit, but either way.

The net gain is that k8s plugins -- or anything without a fully set up pod namespace, including sysadmins who ssh in to debug -- can resolve services that live inside k8s itself. 

A quick test by manually adding 10.3.0.10 (our chosen IP for DNS service) to /etc/resolv.conf seems to work without a hitch. So this bug is asking the question of "should we canonicalize that"?

\* For part 2, making `svc.cluster.local` a configuration option in the script might be worth thinking about a t the same time -- eg, `svc.kubernetes.my-awesome-company.com`

**Comment by pieterlange** [comment]
I can imagine this having all kinds of cyclic errors. -1
I suggest running a "debug pod" and just `kubectl exec`'ing into that to do your kubernetes debugging.

Making the cluster domain configurable should be a separate issue.

**Comment by aaronlevy** [comment]
Yeah, this would be fragile from a bootstrap perspective (chicken/egg). It would have to be that the host resolv.conf was only changed after the clusterDNS started working.

Also, the configurable cluster dns should be a separate issue if that's a feature you would like. Once upon a time we did have it as a configurable, but it just caused more confusion than it was worth (whatever you choose needs to be configured correctly in the addon itself, all kubelets, and have api-server TLS assets with correct SANs). People started standardizing on cluster.local and we just kept with that.
